src/interpret.c

```diff
@@ -4221,6 +4221,15 @@ class Interpreter : public Visitor
 
     void visit(BinAssignExp *e)
     {
+        if (goal == ctfeNeedLvalue)
+        {
+            Expression *e1 = e->e1;
+            while (e->e1->op == TOKcast)
+                e1 = ((CastExp *)e1)->e1;
+            result = interpret(e1, istate, goal);
```

Comment:
I don't think this is right, it will not evaluate the RHS at all.
eg
```d
int ctfetest()
{
    int x;
    ++(x += 1);
    return x == 2;
}

void main()
{
    enum v = ctfetest();
    assert(v == ctfetest());
}
```